Lib/test/test_readline.py

```diff
@@ -349,6 +350,31 @@ def test_history_size(self):
             self.assertEqual(len(lines), history_size)
             self.assertEqual(lines[-1].strip(), b"last input")
 
+    @requires_working_threading()
+    @requires_gil_enabled()
+    def test_gh123321_threadsafe(self):
+        """gh-123321: readline should be thread-safe and not crash"""
+        script = textwrap.dedent(r"""
+        import threading
```

Comment:
With one more indentation level, it will be perfect :-)

Reply:
I like it when people care about the code quality :-)